packages/visual-group/src/encoder/cartesian-encoder.js

```diff
@@ -121,9 +121,18 @@ export default class CartesianEncoder extends VisualEncoder {
                     fieldArr.forEach((field, axisIndex) => {
                         const key = !axisType ? `0${cIdx}${axisIndex}` : `${rIdx}0${axisIndex}`;
                         const dom = unitDomains[`${rIdx}${cIdx}`];
+                        const typeOfField = field.subtype();
+                        const sortingDetails = field.data()._sortingDetails;
+                        const sortingDetailsFlattened = [].concat(...sortingDetails);
+
                         if (dom && Object.keys(dom).length !== 0) {
                             domains[axisType][key] = unionDomain([(domains[axisType] && domains[axisType][key]) || [],
-                                dom[`${field}`]], field.subtype());
+                                dom[`${field}`]], typeOfField);
+                            /* Sort categorical fields to ensure consistency across all rows
+                            only if field is categorical and is not explicitily sorted by user */
+                            if (typeOfField === CATEGORICAL && !sortingDetailsFlattened.includes(field.oneVar())) {
```

Comment:
Here sorting is happening multiple times for same field, which can be done only once.

Reply:
Moved the sorting out